libs/workspace-manager/src/lib/features/filesystem-navigator/filesystem-navigator.component.ts

```diff
@@ -0,0 +1,97 @@
+import { AsyncPipe, NgForOf, NgIf } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  EventEmitter,
+  inject,
+  Input,
+  OnChanges,
+  Output,
+} from '@angular/core';
+import { MatDividerModule } from '@angular/material/divider';
+import { MatIconModule } from '@angular/material/icon';
+import { MatListModule } from '@angular/material/list';
+import { Directory } from '@angular-cli-gui/shared/data';
+import { BehaviorSubject, Observable, of, switchMap, take } from 'rxjs';
+
+import { WorkspaceManagerApiService } from '../../data-access/workspace-manager-api.service';
+
+import { FilesystemNavigatorToolbarComponent } from './filesystem-navigator-toolbar/filesystem-navigator-toolbar.component';
+import { FilesystemNavigatorUtilsService } from './services/filesystem-navigator-utils.service';
+
+@Component({
+  selector: 'cli-filesystem-navigator',
+  standalone: true,
+  imports: [
+    MatListModule,
+    NgForOf,
+    NgIf,
+    MatDividerModule,
+    MatIconModule,
+    FilesystemNavigatorToolbarComponent,
+    AsyncPipe,
+  ],
+  templateUrl: './filesystem-navigator.component.html',
+  styleUrls: ['./filesystem-navigator.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FilesystemNavigatorComponent implements OnChanges {
+  private readonly fsNavigatorUtilsService = inject(
+    FilesystemNavigatorUtilsService
+  );
+  private readonly workspaceManagerApiService = inject(
+    WorkspaceManagerApiService
+  );
+
+  @Input() path: string | null = null;
+
+  @Output() pathChange = new EventEmitter<string>();
+
+  private readonly _isAngularWorkspace$ = new BehaviorSubject<boolean>(false);
+  readonly isAngularWorkspace$ = this._isAngularWorkspace$.asObservable();
+  directories$: Observable<Directory[]> = of([]);
+
+  ngOnChanges(): void {
+    if (this.path) {
+      this.directories$ = this.workspaceManagerApiService.getDirectoriesInPath(
+        this.path
+      );
```

Comment:
directories.next instead of re-assigning. this format of managing directories will result in memory leaks